content/developer/addons/addon-quickstart.md

```diff
@@ -0,0 +1,233 @@
+
+Open the [core repo](http://github.com/vanilla/vanilla) in an IDE and use the project search functionality to locate instances of `fireEvent()`. Many doc blocks show what events are fired within their methods.
+
+Alternatively, search for `_handler` in the [addons repo](http://github.com/vanilla/addons) to see examples and common uses.
+
+Lastly, [find an addon](https://open.vanillaforums.com/addons) that does something similar to what you're trying to do, and check out its code. Maybe between 3 or 4 addons you can find a good portion of what you're trying to accomplish.
+
+### 4. Write a plugin in your addon
+
+After creating your `addon.json` file, you will want to make your a file. Not all addons need to define a plugin. Themes generally don't. Inside your addon's folder, create a file that includes the slug name (e.g. `class.fancyplugin.php`). Next open the file and create a class extending `Gdn_Plugin`, and name the class using the slug name of your plugin followed by the word 'Plugin' to avoid namespace conflicts.
+
+```
+class FancyPlugin extends Gdn_Plugin {
+
+}
+```
+
+Inside the class, write some event handlers. Most handlers are written like this:
+
+```php
+public function someController_someEvent_handler($sender, $args) {
+   // Do something.
+}
+```
+
+Use the `$sender` variable to call methods from the controller or model that triggered your code. 
+
+Use the `$args` variable to change parameters or detect certain situations. This is where you'll want to start digging into the finer points of [extending Vanilla with Event Handlers documentation](/developer/addons/events-and-handlers).
+
+You can also use normal methods to reuse code. For instance, maybe you need 2 different hooks to do the same thing:
+
+```php
+public function discussionsController_someEvent_handler($sender, $args) {
+   $this->commonMethod($sender, $args);
+}
+
+public function categoriesController_someOtherEvent_handler($sender, $args) {
+   $this->commonMethod($sender, $args);
+}
+
+public function commonMethod($sender, $args) {
+   // Do something for both the events above so we keep things DRY.
+}
+```
+
+Need a simple settings page? See the [ConfigurationModule](https://github.com/vanilla/vanilla/blob/master/applications/dashboard/modules/class.configurationmodule.php) and how it is used in other plugins for a quick & easy way of making one.
+
+If you need to, you can [override functions from core](/developer/addons/function-overrides). Do this conservatively and carefully.
+
+### 5. Add CSS or javascript
+
+By convention, place CSS files in a folder named `design`, Javascript files in a folder named `js`, views or templates in a folder named `views`.
+
+To include a CSS file in all non-Dashboard pages:
+
+```php
+public function assetModel_styleCss_handler($sender) {
+   $sender->addCssFile('fancy.css', 'plugins/fancyaddon');
+}
+```
+
+Note the second parameter is the folder.
+
+To include a CSS file in a specific page (e.g. the Discussions list):
```

Comment:
This would target a specific controller not a specific page.
One controller can have multiple pages.